server/src/user-house/test/user-house.e2e-spec.ts

```diff
@@ -88,4 +88,231 @@ describe('UserHouseController (e2e)', () => {
+        .post('/auth/login')
+        .send({
+          email: 'bob@example.com',
+          password: 'password',
+        });
+
+      token = await loginUser.body.accessToken;
+
+      const houseId = 107; // Logged-in user house
+
+      await request(app.getHttpServer())
+        .put('/user-houses/admin')
+        .send({
+          userId,
+          houseId,
+          isAdmin: false,
+        })
+        .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+        .set('x-csrf-token', csrfToken)
+        .expect(403);
+    });
+
+    it('should return 404 if userHouse does not exist', async () => {
+      // Log in as an admin
+      const loginUser = await request(app.getHttpServer())
+        .post('/auth/login')
+        .send({
+          email: 'alice@example.com',
+          password: 'password',
+        });
+
+      token = await loginUser.body.accessToken;
+
+      // Use IDs that are unlikely to exist
+      const houseId = 10000;
+      const userId = 10000;
+
+      await request(app.getHttpServer())
+        .put('/user-houses/admin')
+        .send({
+          userId,
+          houseId,
+          isAdmin: false,
+        })
+        .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+        .set('x-csrf-token', csrfToken)
+        .expect(404);
+    });
+
+    it('should return 400 if user is trying to update their own status', async () => {
+      // Log in as an admin
+      const loginUser = await request(app.getHttpServer())
+        .post('/auth/login')
+        .send({
+          email: 'alice@example.com',
+          password: 'password',
+        });
+
+      token = await loginUser.body.accessToken;
+
+      await request(app.getHttpServer())
+        .put('/user-houses/admin')
+        .send({
+          userId, // Assuming this user matches the loggedIn user
+          houseId,
+          isAdmin: false,
+        })
+        .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+        .set('x-csrf-token', csrfToken)
+        .expect(400);
+    });
+
+    it('should return 200 and change admin status to true', async () => {
+      // Log in as an admin
+      const loginUser = await request(app.getHttpServer())
+        .post('/auth/login')
+        .send({
+          email: 'diana@example.com',
+          password: 'password',
+        });
+
+      token = await loginUser.body.accessToken;
+
+      const houseId = 109; // Assuming this house is the logged-in user house
+      const userId = 105; // Assuming this user is the member of the logged-in user house
+
+      const response = await request(app.getHttpServer())
+        .put('/user-houses/admin')
+        .send({
+          userId,
+          houseId,
+          isAdmin: true,
+        })
+        .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+        .set('x-csrf-token', csrfToken)
+        .expect(200);
+
+      expect(response.body.isAdmin).toBe(true);
+    });
+  });
```

Comment:
Yeah I admit this is a little confusing, but once you login in with other users, alice is no longer a logged-in user, so I need to pick a user that is good for a test.

The reason why admins should not change their own status is because only admins can see this page.
if they chagned their own status, they should not see this page anymore. This means there would be a situation, admins are still seeing this page despite being no longer an admin, and I think it' not good.

Reply:
It's actually hard to change because beforeAll runs only before the whole test of this file starts and once the first test runs and the logged-in user is already overwritten by someone else.

Reply:
Fixed!
https://github.com/HAK2024/fairy-share/commit/2c6afcb427bfb3f1ff8752843337c0184500722b